src/scattdata.cpp

```diff
@@ -0,0 +1,937 @@
+#include "scattdata.h"
+
+namespace openmc {
+
+//==============================================================================
+// ScattData base-class methods
+//==============================================================================
+
+void
+ScattData::base_init(int order, int_1dvec& in_gmin, int_1dvec& in_gmax,
+                        double_2dvec& in_energy, double_2dvec& in_mult)
```

Comment:
Should this just be a constructor?

Reply:
I've had real difficulties getting base_init, base_combine, combine, init, and convert_legendre_to_tabular to simply be constructors.  In the end I went with this way since nothing above Mgxs should ever need to access these guys.  It also made the code a bit more C++-fancy with the use of delegating constructors which I actually find a bit less clear what is going on.